inc/Engine/Admin/PerformanceMonitoring/Jobs/Manager.php

```diff
@@ -201,6 +230,46 @@ public function get_optimization_type_from_row( $row ): string {
 		return $this->optimization_type;
 	}
 
+	/**
+	 * Validate job data to ensure it has required fields with valid values.
+	 *
+	 * @param array  $job_details Details related to the job.
+	 * @param object $row_details Details related to the row.
+	 * @return string Empty string if valid, error message if invalid.
+	 */
+	private function validate_job_data( array $job_details, $row_details ): string {
```

Comment:
Same thing with this as well. If `status` is `failed`, we don't need further checks.